### PR TITLE
1627: Add, populate and use case_number on Case

### PR DESCRIPTION
**Pull request by ahernp**
Trello card [#1627](https://trello.com/c/ihmc4sco/1627-add-custom-case-id-to-case-model).

### CODE REVIEW

File: accessibility_monitoring_platform/apps/cases/migrations/0093_populate_case_numbers.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Generated by Django 5.0.4 on 2024-04-16 13:16
+
+from django.db import migrations
+
+
+def populate_case_numbers(apps, schema_editor):  # pylint: disable=unused-argument
+    Case = apps.get_model("cases", "Case")
+    for case in Case.objects.all():
+        case.case_number = case.id
+        case.save()
```
Comment (richardbored) [inline accessibility_monitoring_platform/apps/cases/migrations/0093_populate_case_numbers.py:10]: I think we need to give the case number their correct sequential number. Can we rely on the iterator to show them in the correct order?

File: accessibility_monitoring_platform/apps/common/migrations/0021_populate_issue_numbers.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Generated by Django 5.0.4 on 2024-04-17 08:16
+
+from django.db import migrations
+
+
+def populate_issue_numbers(apps, schema_editor):  # pylint: disable=unused-argument
+    IssueReport = apps.get_model("common", "IssueReport")
+    for issue_report in IssueReport.objects.all():
+        issue_report.issue_number = issue_report.id
+        issue_report.save()
```
Comment (richardbored) [inline accessibility_monitoring_platform/apps/common/migrations/0021_populate_issue_numbers.py:10]: Are issue ids user facing? If not, Im less concerned that these are 'fixed'

File: accessibility_monitoring_platform/apps/cases/migrations/0093_populate_case_numbers.py (hunk 1 above)
Comment (ahernp) [reply accessibility_monitoring_platform/apps/cases/migrations/0093_populate_case_numbers.py:10]: I do not understand your comment.

What do you mean by 'correct' if not the `Case.id` which has been used up to now?

The order of cases returned in this loop makes no difference.

Chris said that he doesn't want the numbers of existing cases to change.

File: accessibility_monitoring_platform/apps/common/migrations/0021_populate_issue_numbers.py (hunk 2 above)
Comment (ahernp) [reply accessibility_monitoring_platform/apps/common/migrations/0021_populate_issue_numbers.py:10]: The issue report ids are user facing: They appear in the admin when we go through the user feedback and in my email to the users confirming that their feedback has been received.

However, they have no semantic meaning. They merely need to be unique.

Comment (richardbored) [reply accessibility_monitoring_platform/apps/common/migrations/0021_populate_issue_numbers.py:10]: I just saw that haha

**Review by richardbored — APPROVED** [review]

**Pull request merged**